aliyun/src/main/java/org/apache/iceberg/aliyun/DefaultAliyunClientFactory.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aliyun;
+
+import com.aliyun.oss.OSS;
+import com.aliyun.oss.OSSClientBuilder;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+public class DefaultAliyunClientFactory implements AliyunClientFactory {
```

Comment:
When we were writing `AwsClientFactory` we had `AwsClientFactories` to vend a default client factory implementation, and that was because it seems like Iceberg does not expose the default implementation as a public class for other classes like `LocationProvider`, etc. I wonder if we should follow the same pattern here.

https://github.com/apache/iceberg/blob/master/aws/src/main/java/org/apache/iceberg/aws/AwsClientFactories.java

Reply:
Yes, I agree.  As we will also introduce many different factories for aliyun clients,   it's good to have a proxy to delegate to create the factories.